### PR TITLE
Containerize

### PR DESCRIPTION
**Pull request by Snafkin547**
Dockerized the app and changed instructions in ReadMe accordingly:

- Users now can start using it by just cloning and building/composing images/container
- Port used is changed from 8080 to 3000
- Local machine and quick start instructions are now consolidated to avoid duplication
- I did not change the instruction for EC2

### CODE REVIEW

File: db/WiredTiger [hunk 1]
```diff
@@ -0,0 +1,2 @@
+WiredTiger
```
Comment (gregfrasco) [inline db/WiredTiger]: we can add db folder to git ignore

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+version: '3'
+services:
+  web-mpc:
+    container_name: web-mpc
+    restart: always
+    build: .
+    ports:
+      - "3000:8080"
```
Comment (gregfrasco) [inline docker-compose.yml]: probably want to key the same port
```suggestion
      - "8080:8080"
```

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    volumes:
+      - ./:/code
+    depends_on:
+      - mongo
+
+  mongo:
+    container_name: mongo
+    image: mongo:3.4
+    ports:
+      - "27018:27017"
```
Comment (gregfrasco) [inline docker-compose.yml]:
```suggestion
      - "27017:27017"
```

File: package.json [hunk 4]
```diff
@@ -23,7 +23,8 @@
     "le-acme-core": "^2.1.3",
     "mathjs": "^5.0.4",
     "mocha": "^6.1.4",
-    "mongoose": "^4.11.1",
+    "mongodb": "^4.9.0",
```
Comment (gregfrasco) [inline package.json]: don't need to add mongodb here

File: server/config/testing.json [hunk 5]
```diff
@@ -0,0 +1,24 @@
+{
```
Comment (gregfrasco) [inline server/config/testing.json]: dont need this here

File: server/models/models.js [hunk 6]
```diff
@@ -8,7 +8,7 @@ const mongoose = require('mongoose');
 
 (async function () {
   try {
-    await mongoose.connect('mongodb://localhost/aggregate', { useMongoClient: true });
+    await mongoose.connect('mongodb://mongo:27017/aggregate', { useMongoClient: true });
```
Comment (gregfrasco) [inline server/models/models.js]: could we convert this to an env var so it can be changed to which env you are using

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+# Define directory inside the container
+WORKDIR /usr/src/app
+
+# Copying the local directory to the container directory(/usr/src/app)
+COPY . .
+
+RUN git submodule init jiff
+RUN git submodule update
+
+#Install app dependecies (wildcard ensures both package/package-loc.json are copied)
```
Comment (gregfrasco) [inline Dockerfile]:
```suggestion
#Install app dependecies (wildcard ensures both package/package-lock.json are copied)
```

File: Dockerfile [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+RUN npm install
```
Comment (gregfrasco) [inline Dockerfile]: Swicth these out for npm ci
```suggestion
RUN npm ci
```

File: db/WiredTiger (hunk 1 above)
Comment (Snafkin547) [reply db/WiredTiger]: Added to gitignore

File: server/statics.js [hunk 9]
```diff
@@ -38,4 +38,5 @@ module.exports = function (app) {
   app.use('/jiff', express.static(__dirname + '/../jiff/dist'));
   app.use('/jiff/ext', express.static(__dirname + '/../jiff/lib/ext'));
   app.use('/bignumber.js', express.static(__dirname + '/../jiff/node_modules/bignumber.js'));
+  console.log('here',__dirname);
```
Comment (gregfrasco) [inline server/statics.js]: remove this

**Review by gregfrasco — APPROVED** [review]

**Pull request merged**